fontbakery-check-ttf.py

```diff
@@ -685,6 +720,149 @@ def ttf_file(f):
     log_results("fsType is zero.")
 
     # ----------------------------------------------------
+    fb.new_check("Assure valid format for the"
+                 " main entries in the name table.")
+    # Each entry in the name table has a criteria for validity and
+    # this check tests if all entries in the name table are
+    # in conformance with that. This check applies only
+    # to name IDs 1, 2, 4, 6, 16, 17, 18.
+    # It must run before any of the other name table related checks.
+
+    failed = False
+    # The font must have at least these name IDs:
+    for nameId in [NAMEID_FONT_FAMILY_NAME,
+                   NAMEID_FONT_SUBFAMILY_NAME,
+                   NAMEID_FULL_FONT_NAME,
+                   NAMEID_POSTSCRIPT_NAME,
+                   NAMEID_TYPOGRAPHIC_FAMILY_NAME,
+                   NAMEID_TYPOGRAPHIC_SUBFAMILY_NAME]:
```

Comment:
This is wrong; the 4 Regular/Italic/Bold/Bold Italic styles don't have NAMEID_TYPOGRAPHIC_FAMILY_NAME or NAMEID_TYPOGRAPHIC_SUBFAMILY_NAME